### ISSUE TITLE
bug(spectate race condition): two people joining the last spot in a game at the same time [need verification]

### ISSUE THREAD

**Issue by pikulet** [r3ndd/BeyondMafia-Integration#59]
by [Rutab](https://beyondmafia.com/community/forums/thread/vYPKWTAh7)

It said there was one spot left in a game of TSIB. I joined it and someone else did too while my computer was loading into it. I think if two people click "join" with one spot left, it's going to give it to the person who connected first. It prevents the latter person from joining as a participant, however I was able to spectate even though the setting was toggled off.

TLDR: Two people joining for a last spot allows one to be a spectator in a game that has spectating turned off.
